platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
+    fn metadata(resource_id: ResourceId) -> Result<Metadata, String>;
+    fn read(resource_id: ResourceId) -> Result<Content, String>;
+    fn remove(resource_id: ResourceId) -> Result<(), String>;
+    // import will decide if it needs to unzip a file. The contents of the url will be extracted and inserted into a database.
```

Comment:
Agreed. I like the clay tablet idea 😎

Reply:
Did you see the implementation of TCP over avian carriers?

Reply:
Hm..... I'm not sure I like this :D Let's think about the use cases we have right now:

1. We have a ZIP file external to the polyPod, whose _contents_ we want to import.

That's all :D Let's think about other potential use cases:

2. We have an arbitrary file external to the polyPod that we want to import (i.e. copy).
3. We have an arbitrary directory external to the polyPod that we want to import

Currently, (1) is possible, via `import_archive`, while (2) and (3) are not.

With this design, (1) and (3) would probably be possible, and (2) with some caveat - depending on what archive formats the polyPod supports, it would either do (2) or (1) - seems **super** error prone. I think explicit would be better, e.g.:

- `import_archive` (gets an archive file)
- `import` (gets a file or directory)

Note that we have no need for `import` (without the `_archive`) yet, but we probably will have, same as for `write`.

Reply:
Did you remember what I said when you created importArchive for the first time? At the end of the day it's import + processing. I don't know if @timoteipalade was going in that direction, but it would make sense to make a smartish function that decides on processing depending on the type of file; smaller API footprint and more adaptable (what if we want to import something compressed with 7z, rar or tgz?)

Reply:
I'd be against anything trying to be smart there - that'd just make the user feel dumb :P What if they do want to store a ZIP file as is, e.g. because they want to know it's meta data, look for steganography, malware or whatever - who are we to determine how it's stored? The user should tell the API what they want IMO.

Reply:
Well, it's for us to decide what an interface can or can't do, according to the use cases we come up with. In that case I'd add some function simply called "store". You said yourself above that something like `extract_and_import` would make sense; maybe calling it that way, or calling it import_and_process or something like that would make sense. If you don't want any smartness, so be it, but specifying the kind of processing (or extracting) it needs in a separate function would make extension very easy.